### PR TITLE
Fixing docker test issues @rev darshshah@

### PR DESCRIPTION
**Pull request by lastmansleeping**

### CODE REVIEW

**Comment by lastmansleeping** [comment]
@darshshah  The PR has the minor syntax changes for fixing the tests.

Still trying to see why the serving tests are failing on docker.

**Comment by darshshah** [comment]
Can you write a high-level summary on what was broken and how these changes will fix it?

File: python/ml4ir/applications/ranking/tests/test_base.py [hunk 1]
```diff
@@ -71,6 +72,10 @@ def tearDown(self):
         # Delete other temp directories
         file_io.rm_dir(os.path.join(self.root_data_dir, "csv", "tfrecord"))
 
+        # Clear memory
+        tf.keras.backend.clear_session()
```
Comment (lastmansleeping) [inline python/ml4ir/applications/ranking/tests/test_base.py:76]: @darshshah Adding this for better clean up. Clears any memory held by keras. But this will not help for tf2.0 sadly. I still think it is a clean practice to have this for when we upgrade tensorflow.

File: python/ml4ir/base/features/feature_fns/categorical.py [hunk 2]
```diff
@@ -204,7 +205,7 @@ def categorical_embedding_with_vocabulary_file(feature_tensor, feature_info):
     vocabulary_df = file_io.read_df(feature_layer_info["args"]["vocabulary_file"])
     vocabulary_keys = vocabulary_df["key"].values
     vocabulary_ids = (
-        vocabulary_df["id"].values if "id" in vocabulary_df else range(len(vocabulary_keys))
+        vocabulary_df["id"].values if "id" in vocabulary_df else list(range(len(vocabulary_keys)))
```
Comment (lastmansleeping) [inline python/ml4ir/base/features/feature_fns/categorical.py:208]: @darshshah This was a syntax error. Had to be a list, but was range instead. The reason it wasn't caught is because it works for training and testing and only throws error when saving.

File: python/ml4ir/base/model/relevance_model.py [hunk 3]
```diff
@@ -348,17 +348,16 @@ def save(
...
         """
         Save model with custom signatures
 
         Currently supported
         - signature to read TFRecord SequenceExample inputs
         """
-        saved_model.save(
-            self.model,
-            export_dir=os.path.join(model_file, "tfrecord"),
+        self.model.save(
```
Comment (lastmansleeping) [inline python/ml4ir/base/model/relevance_model.py:359]: Unrelated syntax change. Should be hitting the same API. I read that some memory improvements can be expected. But I didn't observe any. Still clean to keep it this way because both save and load are not keras model APIs - so there's no mismatch.

File: python/ml4ir/base/features/feature_fns/categorical.py [hunk 4]
```diff
@@ -165,7 +165,8 @@ def get_config(self):
         config = super(VocabLookup, self).get_config()
         config.update(
             {
-                "vocabulary_list": self.vocabulary_list,
+                "vocabulary_keys": self.vocabulary_keys,
```
Comment (lastmansleeping) [inline python/ml4ir/base/features/feature_fns/categorical.py:168]: @darshshah Syntax error fix.

**Comment by lastmansleeping** [comment]
> Can you write a high-level summary on what was broken and how these changes will fix it?

@darshshah Added comments. Very trivial changes really.

You can run the tests by `python3 -m pytest` to verify.

**Comment by darshshah** [comment]
Looks good. I'll let the CI run the tests are verify ;)
